internal/query/map.go

```diff
@@ -1,56 +1,117 @@
 package query
 
 import (
+	"errors"
 	"fmt"
 	"strings"
 )
 
 // GetMap returns a map, which satisfies conditions.
-func GetMap(query map[string][]string, key string) (dicts map[string]interface{}, exists bool) {
+func GetMap(query map[string][]string, key string) (dicts map[string]interface{}, err error) {
 	result := make(map[string]interface{})
+	getAll := key == ""
+	var allErrors = make([]error, 0)
 	for qk, value := range query {
-		if isKey(qk, key) {
-			path, err := parsePath(qk, key)
+		kType := getType(qk, key, getAll)
+		switch kType {
+		case "filtered_unsupported":
```

Comment:
I would suggest to make an "enum" for that `kType` values. And I suppose, they could be numeric (that would speed up the algo unnoticeably 🙈).